jdaviz/conftest.py

```diff
@@ -226,6 +226,18 @@ def mos_spectrum1d():
     return Spectrum1D(spectral_axis=spec_axis, flux=flux)
 
 
+@pytest.fixture
+def spectrum2d(flux_unit=u.Unit('MJy'), spectral_unit=u.um):
```

Comment:
So the reason why I left these args is two thoughts:

1. If you look above, I implemented this for the cube fixture because one test I added specifically needed the cube in a different unit. And rather than copying a whole new fixture, I just added unit args. So there's a precedent for it
2. Because of the upcoming unit conversion work, combined with the uneven wavelength axes work coming up soon, I figure it will be useful in those tests. 🤷

Reply:
I actually don't think you can define a fixture like that, at least not from the examples I see in https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#fixture-parametrize . Is there a working example from somewhere you are basing this implementation from?

Reply:
Yep! You can see it in action on main here: https://github.com/spacetelescope/jdaviz/blob/f8b31919df8ff91d3db90c9a644b5a276a53295f/jdaviz/configs/specviz/plugins/line_analysis/tests/test_lineflux.py#L77

I'm basically parameterizing a bunch of different test cases I define higher up and "on the fly" generate different cubes of different flux units: https://github.com/spacetelescope/jdaviz/blob/f8b31919df8ff91d3db90c9a644b5a276a53295f/jdaviz/configs/specviz/plugins/line_analysis/tests/test_lineflux.py#L13-L21

This also, separately, allows me to generate new pytest fixtures dynamically and easily without having to duplicate code that all looks VERY similar: https://github.com/spacetelescope/jdaviz/blob/f8b31919df8ff91d3db90c9a644b5a276a53295f/jdaviz/conftest.py#L182-L206

Reply:
Still not sure about those args in this fixture though.

Reply:
Fair point! I removed it in https://github.com/spacetelescope/jdaviz/pull/2023/commits/c2ce1007401cf7c6bd3f3e06fffea166f45a5656 and we can revisit if it ever becomes useful in the future